core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -308,6 +308,9 @@ sealed abstract class IO[+E, +A] { self =>
   final def ensuring(finalizer: IO[Nothing, Unit]): IO[E, A] =
     new IO.Ensuring(self, finalizer)
 
+  final def on(ec: ExecutionContext): IO[E, A] =
```

Comment:
It might make sense to also add `forkOn`

```scala
final def forkOn(ec: ExecutionContext): IO[Nothing, Fiber[E, A]] =
  on(ec).fork
```

Reply:
I like this one. The implementation perhaps feels a bit weird because it might not be necessary to shift back to the default pool just to product the result.

Also, what do we want the order of operations to be? Shift, fork, then shift back? Or fork, shift, and shift back?

Reply:
Now that I think about it, shift and then fork doesn't make any sense because fork submits the task on the default pool.

Reply:
> Users of this function would need to be aware that inserting an IO.shift would shift execution to the RTS pool.

Yes, but we might want to change that in the future, I.e. make fiber remember its execution context to have better compatibility with `cats-effect` and libraries that want the user to supply an explicit ec - currently at least fs2, http4s, doobie.

Reply:
> The implementation perhaps feels a bit weird because it might not be necessary to shift back to the default pool just to product the result.

Ah, yes, that's redundant. It should probably be just:

```scala
(IO.shift(ec) *> self).fork
```